packages/service/index.js

```diff
@@ -59,15 +58,58 @@ async function platformaticService (app, opts, toLoad = []) {
     }
   }
 
-  // TODO apparently c8 is not able to mark
-  // this as covered even if it is
-  /* c8 ignore next 7 */
-  if (Array.isArray(opts.plugin)) {
-    for (const plugin of opts.plugin) {
-      await loadPlugin(app, opts, plugin)
+  if (opts.plugins) {
+    // if we don't have a fullPath, let's assume we are in a test and we can use the current working directory
+    const configPath = app.platformatic.configManager.fullPath || process.cwd()
+    const tsConfigPath = join(dirname(configPath), 'tsconfig.json')
+    /* c8 ignore next 21 */
+    if (await isFileAccessible(tsConfigPath)) {
+      const tsConfig = JSON.parse(await readFile(tsConfigPath, 'utf8'))
+      const outDir = resolve(dirname(tsConfigPath), tsConfig.compilerOptions.outDir)
+      opts.plugins.paths = opts.plugins.paths.map((plugin) => {
+        if (typeof plugin === 'string') {
+          return getJSPluginPath(configPath, plugin, outDir)
+        } else {
+          return {
+            path: getJSPluginPath(configPath, plugin.path, outDir),
+            options: plugin.options
+          }
+        }
+      })
+    } else {
+      for (const plugin of opts.plugins.paths) {
+        const path = typeof plugin === 'string' ? plugin : plugin.path
+        if (path.endsWith('.ts')) {
+          throw new Error(`Cannot load plugin ${path}, tsconfig.json not found`)
+        }
+      }
+    }
+
+    // if not defined, we defaults to true (which can happen only if config is set programmatically,
+    // that's why we ignore the coverage of the `undefined` case, which cannot be covered in cli tests)
+    // all individual plugin hot reload settings will be overloaded by global hot reload
+    /* c8 ignore next 1 */
+    const hotReload = opts.plugins.hotReload ?? opts.plugins.hotReload !== false
```

Comment:
Isn't `opts.plugins.hotReload !== false` guaranteed to be `true`? If it were `false`, the `??` wouldn't reach that branch.

Reply:
You are right! Thanks for catching it.

Reply:
```
const hotReload = opts.plugins.hotReload ?? opts.plugins.hotReload !== false
```
should it be?

```
const hotReload = opts.hotReload ?? opts.plugins.hotReload !== false

```

or that global hotReload is going to be dropped in favor of `opts.plugins.hotReload` ref https://github.com/platformatic/platformatic/pull/679

Reply:
It seems the test in https://github.com/platformatic/platformatic/pull/679/files is not really working as expected because this branch is passing and it shouldn't if the feature was implemented correctly.

There shouldn't be any global `hotReload` config, but rather it should be set inside the `plugins` key in the config and then overridden there. I'll add some more tests here.

Reply:
fixed